tests/projects/test_docker_projects.py

```diff
@@ -258,12 +258,12 @@ def test_docker_user_specified_env_vars(volumes, environment, expected, os_envir
 
     if "should_crash" in expected:
```

Comment:
```suggestion
    monkeypatch.setenvs(os_environ)
    if "should_crash" in expected:
```

I think we can move `monkeypatch.setenvs(os_environ)` here